robot-name/robot_name.go

```diff
@@ -0,0 +1,83 @@
+// Manage robot factory settings.
+package robotname
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+// Robot has a name
+type Robot struct {
+	name string
+}
+
+// All the existing robot names so we can check for unique names
+var robots = make(map[string]Robot)
+
+// Name returns a robot's name in the format of two uppercase letters
+// followed by three digits. If the robot doesn't have a name already,
+// a random name is generated. Names must be random (no predictable pattern)
+// and unique (no duplicate names).
+func (r *Robot) Name() (string, error) {
+	if r.name == "" {
+		newName, error := GenerateUniqueName()
+		if error != nil {
+			return "", error
+		}
+		r.name = newName
+		robots[newName] = *r
+	}
+	return r.name, nil
+}
+
+// Reset clears the name of a robot so the next time a name is requested a
+// new random name is generated.
+func (r *Robot) Reset() {
+	r.name = ""
+}
+
+// Maximum number of unique name possible (26x26x1000)
+var MAX_NAMES = 676000
```

Comment:
This should be declared as a constant. Also, I don't think the name is adequate. Even with constants, the standard library uses camel case. Then, the first letter could be lower case for non-exported constants or upper case for those variables that are used outside the package.